sw/device/mask_rom/boot.md

```diff
@@ -0,0 +1,110 @@
+# Pseudo-code for Secure Boot Process.
+
+This file should be read in conjunction with secure boot specification. Some
+references to that document are included.
+
+
+1. Power on (entirely in hardware) Open Q: Whether SW has to configure PMP
+  initial region.
+
+2. Execution Begins
+  - CRT Startup Code (ASM)
+    - Clean Slate (SRAM, Registers)
+    - Well-defined interrupt/exception handler
+        Open Q: Do we want to enable interrupts in mask rom?
+    - Jump into C code (`boot`):
+
+```c
+void boot(void) {
+  // - Clean all other hardware state (using DIF Reset functions)
+  clean_device_state();
```

Comment:
sorry this is something I've discussed with @gkelly / @moidx but still have not transferred into documentation.
There should be a correlation between boot-state and what memories need to be wiped, and what memory scrambling needs to be setup.  I guess that would go here as well?

Reply:
Based on the meeting, `clean_device_state` is about making sure there's nothing in an improper state. Setting up scrambling should happen pretty early, but probably as a separate step.